projects/step-frontend/src/lib/modules/_common/components/sidebar/sidebar.component.ts

```diff
@@ -0,0 +1,136 @@
+    // Main Menus
+    this._viewRegistryService.registerCustomMenuEntry('Automation', 'automation-root', 'glyphicon glyphicon-play');
+    this._viewRegistryService.registerCustomMenuEntry('Execute', 'execute-root', 'glyphicon glyphicon-tasks');
+    this._viewRegistryService.registerCustomMenuEntry('Status', 'status-root', 'glyphicon glyphicon-ok');
+
+    // Sub Menus Automation
+    this._viewRegistryService.registerCustomMenuEntry(
+      'Keywords',
+      'functions',
+      'glyphicon glyphicon-record',
+      'automation-root'
+    );
+    this._viewRegistryService.registerCustomMenuEntry('Plans', 'plans', 'glyphicon glyphicon-file', 'automation-root');
+    this._viewRegistryService.registerCustomMenuEntry(
+      'Parameters',
+      'parameters',
+      'glyphicon glyphicon-list-alt',
+      'automation-root'
+    );
+    // Sub Menus Execute
+    this._viewRegistryService.registerCustomMenuEntry(
+      'Executions',
+      'executions',
+      'glyphicon glyphicon-tasks',
+      'execute-root'
+    );
+    this._viewRegistryService.registerCustomMenuEntry(
+      'Scheduler',
+      'scheduler',
+      'glyphicon glyphicon-time',
+      'execute-root'
+    );
+    // Sub Menus Status
+    this._viewRegistryService.registerCustomMenuEntry(
+      'Agents',
+      'gridagents',
+      'glyphicon glyphicon-briefcase',
+      'status-root'
+    );
+    this._viewRegistryService.registerCustomMenuEntry(
+      'Agent tokens',
+      'gridtokens',
+      'glyphicon glyphicon-tag',
+      'status-root'
+    );
+    this._viewRegistryService.registerCustomMenuEntry(
+      'Token Groups',
+      'gridtokengroups',
+      'glyphicon glyphicon glyphicon-tags',
+      'status-root'
+    );
+    this._viewRegistryService.registerCustomMenuEntry(
+      'Quota Manager',
+      'gridquotamanager',
+      'glyphicon glyphicon-road',
+      'status-root'
+    );
+    this.mainMenuEntriesAmount = _viewRegistryService.getCustomMainMenuEntries().length - 1;
+  }
+
+  ngAfterViewInit(): void {
+    let initialViewName = this._viewStateService.getViewName();
+    if (initialViewName) {
+      let initiallyExpandedMainMenu = this._viewRegistryService.getMainMenuNameOfSubmenu(initialViewName!);
+      if (initiallyExpandedMainMenu) {
+        this.openMainMenu(initiallyExpandedMainMenu);
+      }
+    }
+  }
+
+  private openMainMenu(mainMenuName: string): void {
+    for (let checkbox of this.mainMenuCheckBoxes!) {
+      let name = checkbox.nativeElement.getAttribute('name');
+      if (name === mainMenuName) {
+        checkbox.nativeElement.setAttribute('checked', true);
+      }
+    }
+  }
+
+  public navigateTo(viewId: string): void {
+    switch (viewId) {
+      case 'home':
+        this._authService.gotoDefaultPage();
+        break;
+      default:
+        this._location.go('#/root/' + viewId);
+        break;
+    }
+  }
+
+  /**
+   * FIXME: communicate open/close state as output to parent component instead of selecting global dom element
```

Comment:
I know that this is something that should not be done in Angular: manipulating the DOM outside of the component from withing the component.

### How do we go on with this?
I suggest that as soon as our root controller is also migrated to NG2+, we can communicate the state of the sidebar (opened or closed) to the parent component as an `@Output` and react to it from there. That would be a clean solution again. I spend some time trying to do this between the sidebar.component and the root AJS controller, but could not make it work and I did not want to waste time.

### What other solutions would be possible?
- *for the left margin of the "main" div*: build a structure that makes it obsolete to adjust the left margin manually, like a parent div with `display:flex` and `flex-direction: row`, and sidebar and main div as flex children, side-by-side. I could make this work on a sandbox page, but failed to apply this to our design successfully. Again, I did not want to waste too much time.
- *for the tenant selector*: in the current solution, the tenant selector is outside of the sidebar component and is manipulated from the outside. So why not put it inside? The reason is that the `step-tenant-selection` component is build in step EE, the `sidebar` is build in step OS. While AJS it is no problem to put an html tag in the template that is not known at compile time, the NG2+ compiler screams. Also AFAIK we do not have a migrated version of Dashlet-Binding that we used in the old version, like ```				<li ng-repeat="entry in viewRegistry.getDashlets('/menu/navbar/right')" >
					<div ng-include="entry.template"></div>
				</li>```

Reply:
Maybe we can open a ticket for this and apply the solution suggested above as soon as the root controller is migrated.